youtube_dlc/extractor/brightcove.py

```diff
@@ -533,6 +538,10 @@ def build_format_id(kind):
                         'format_id': build_format_id('rtmp'),
                     })
                 formats.append(f)
+
+        if sources_num == key_systems_present:
```

Comment:
wouldn't it be better or more elegant to move the raise extractor error to `if not formats` and check there if there have been drm protected streams?

Reply:
I thought about doing that. Are you aware of any better options for determining drm protected streams? The best way I could think of was to simply keep a count of the number sources that are locked and then compare the two.

Reply:
Well. If there aren't any formats but there have been containers (drm) it's obvious that there is a drm related problem. So I think it's best to handle that error inside the `if not formats`. As for determining drm streams. Depending on the manifest it could be extracted. But it's already built-in to filter out drm in the base protocols.